lib/fastly/client.rb

```diff
@@ -158,6 +204,10 @@ def make_params(params)
 
       param_ary.flatten.delete_if { |v| v.nil? }.join('&')
     end
+
+    def should_retry(tries, error_message)
+      (tries < RETRY_COUNT) && ( error_message.downcase.include?('something went wrong') || error_message.downcase.include?('i/o error') )
```

Comment:
Is there a reason to check the error message instead of just retrying while within the RETRY_COUNT threshold?

My concern would be that a change in the error message text would cause the retry to skip.

Reply:
I appreciate this is a shared library. My specific use case is to only retry when the API itself is broken, rather than retry just because a call tried to do something it isn't allowed to do.

I mean one of the great challenges in implementing HTTP retry is knowing if the failure was due to some request that can _never_ succeed versus some request that failed due to a network or a glitch in the API accepting requests. In my experience of using the Fastly API it seems from time to time we get this "500 something wrong" or the "500 I/O Error" error response and that it is temporary and a retry always works.

I'm open to your advice here. Is there a better way to know if a failed response _could_ succeed if it is retried? Or other error responses that we can compare to to recognize a request that can be retried?